tests/check/s3sink.c

```diff
@@ -116,6 +116,30 @@ push_bytes(GstPad *pad, size_t num_bytes, GstFlowReturn expected_ret_code)
 #define PUSH_BYTES(pad, num_bytes) fail_if (!push_bytes(pad, num_bytes, GST_FLOW_OK))
 #define PUSH_BYTES_FAILURE(pad, num_bytes) fail_if (!push_bytes(pad, num_bytes, GST_FLOW_ERROR))
 
+/**
+ * @brief Before pushing buffers/bytes one must send a stream start and a segment declaration
+ * or else critical errors will occur.
```

Comment:
Why was that not needed before?

Reply:
I suspect that more recent gstreamer release builds have enabled [extra checks](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/gstreamer/gst/gstpad.c#L4424) that were previously disabled by default. Without this change, the tests would fail with the following error on ubuntu 20 or later:

```
Unexpected critical/warning: ../gst/gstpad.c:4672:gst_pad_push_data:<'':src> Got data flow before stream-start event
```